include/toml11/conversion.hpp

```diff
@@ -1,9 +1,88 @@
 #ifndef TOML11_CONVERSION_HPP
 #define TOML11_CONVERSION_HPP
 
+#include "find.hpp"
 #include "from.hpp" // IWYU pragma: keep
 #include "into.hpp" // IWYU pragma: keep
 
+#if defined(TOML11_HAS_OPTIONAL)
+
+#include <optional>
+
+namespace toml
+{
+namespace detail
+{
+
+template<typename T>
+inline constexpr bool is_optional_v = false;
+
+template<typename T>
+inline constexpr bool is_optional_v<std::optional<T>> = true;
+
+template<typename T, typename TC>
+void find_member_variable_from_value(T& obj, const toml::basic_value<TC>& v, const char* var_name)
+{
+    if constexpr(is_optional_v<T>)
+    {
+        try
+        {
+            obj = toml::find<typename T::value_type>(v, var_name);
+        }
+        catch(...)
+        {
+            obj = std::nullopt;
+        }
```

Comment:
This does not throw any error if the value exists but the type is different. Let's say we have

```cpp
struct Foo {std::optional<int> x;};
```

and

```toml
[Foo]
x = "bar"
```

We expect that `x` is an integer, but the one who wrote the config mistakenly specifies a string. This results `Foo{.x = std::nullopt}`. Of course this is an option, but we can also throw `type_error` by:

```cpp
if(v.contains(var_name))
{
    obj = toml::get<typename T::value_type>(v.at(var_name));
}
else
{
    obj = std::nullopt;
}
```

Which one do you think better, for example in your use case?

Reply:
Oops, I prefer to throw an exception.  Let me fix this PR shortly.  Thank you!